Orchestrator/Prognetics.Saga.Orchestrator/GuidIdentifierService.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace Prognetics.Saga.Orchestrator;
+
+public class GuidIdentifierService : IIdentifierService
```

Comment:
Isn't it an exaggeration to create a separate class for guide generation?

Reply:
Its abstraction over the generating identifiers, its usefull for unit testing. It can be simpified by injecting delegate instead of service behind the interface like that:
``` csharp
  public delegate IdentifierFactory();

  public SagaEngine(..., IdentifierFactory identifierFactory)

  services.AddSingleton<IdentifierFactory>(() => Guid.NewGuid().ToString());

```